### DISCUSSION TITLE
Why does the mecha-cms/composer plugin delete git files?

### DISCUSSION THREAD

**Discussion by JacopoWolf** [mecha-cms/mecha#291] [Q&A] [answered]
I do understand on a general level that it might be because on a production environment one might not want to have those files around... but why is deleting files like README.md the default?
Shouldn't this be like an opt-in feature to enable on the production server?

**Comment by taufik-nurrohman** [comment, marked as answer]
It was defined in the `.gitattributes` file to exclude those files when downloading GitHub files using `git archive` command. Composer uses that command internally to store the GitHub files in its server, I guess.

Mecha does not use the `README.md` file as its project documentation (to be displayed in the control panel). But instead, it uses `about.page` file, so it can have custom fields.

If you want to develop an extension [^1] or a layout [^2] that includes `README.md` file by default, just remove this line from the `.gitattributes` file in your extension/layout repository:

``` gitattributes
README.md export-ignore
```

Of course, [in the source code](https://github.com/mecha-cms/composer/blob/dcd61aca99c47c3e39a6e6867a77a500b60d9057/src/Plugin.php#L17-L40) there is a list of file and folder names to remove (including `README.md` file). Maybe I should consider removing that list so that Composer can rely solely on file `.gitattributes` in each repository.

[^1]: https://github.com/mecha-cms/x
[^2]: https://github.com/mecha-cms/y

**Reply by JacopoWolf** [reply]
Thank you very much for the reply and for this amazing project.
